addons/mail/static/tests/discuss/core/public_web/sub_channels.test.js

```diff
@@ -271,3 +271,44 @@ test("muted channel hides sub-thread unless channel is selected or thread has un
     );
     await contains(".o-mail-DiscussSidebar-item:contains('New Thread')");
 });
+
+test("show notification when clicking archived/deleted thread", async () => {
```

Comment:
```suggestion
test("show notification when clicking on deleted thread", async () => {
```

Lets keep it simple :) The rest of the test would be much simpler as well. From the frontend point of view, archived or deleted is the same (getOrFetch won't return anything, the toast is displayed).